src/databricks/sqlalchemy/dialect/__init__.py

```diff
@@ -0,0 +1,300 @@
+"""This module's layout loosely follows example of SQLAlchemy's postgres dialect
+"""
+
+import decimal, re, datetime
+from dateutil.parser import parse
+
+from sqlalchemy import types, processors
+from sqlalchemy.engine import default
+from sqlalchemy.exc import DatabaseError
+from sqlalchemy.engine import reflection
+
+from databricks import sql
+
+
+from databricks.sqlalchemy.dialect.base import (
+    DatabricksDDLCompiler,
+    DatabricksIdentifierPreparer,
+)
+from databricks.sqlalchemy.dialect.compiler import DatabricksTypeCompiler
+
+try:
+    import alembic
+except ImportError:
+    pass
+else:
+    from alembic.ddl import DefaultImpl
+
+    class DatabricksImpl(DefaultImpl):
+        __dialect__ = "databricks"
+
+
+class DatabricksDecimal(types.TypeDecorator):
+    """Translates strings to decimals"""
+
+    impl = types.DECIMAL
+
+    def process_result_value(self, value, dialect):
+        if value is not None:
+            return decimal.Decimal(value)
+        else:
+            return None
+
+
+class DatabricksTimestamp(types.TypeDecorator):
+    """Translates timestamp strings to datetime objects"""
+
+    impl = types.TIMESTAMP
+
+    def process_result_value(self, value, dialect):
+        return processors.str_to_datetime(value)
+
+    def adapt(self, impltype, **kwargs):
+        return self.impl
+
+
+class DatabricksDate(types.TypeDecorator):
+    """Translates date strings to date objects"""
+
+    impl = types.DATE
+
+    def process_result_value(self, value, dialect):
+        return processors.str_to_date(value)
+
+    def adapt(self, impltype, **kwargs):
+        return self.impl
+
+
+class DatabricksDialect(default.DefaultDialect):
+    """This dialect implements only those methods required to pass our e2e tests"""
+
+    # Possible attributes are defined here: https://docs.sqlalchemy.org/en/14/core/internals.html#sqlalchemy.engine.Dialect
+    name: str = "databricks"
+    driver: str = "thrift"
+    default_schema_name: str = "default"
+
+    preparer = DatabricksIdentifierPreparer
+    type_compiler = DatabricksTypeCompiler
+    ddl_compiler = DatabricksDDLCompiler
+    supports_statement_cache: bool = True
+    supports_multivalues_insert: bool = True
+    supports_native_decimal: bool = True
+    supports_sane_rowcount: bool = False
+
+    @classmethod
+    def dbapi(cls):
+        return sql
+
+    def create_connect_args(self, url):
+        # TODO: can schema be provided after HOST?
```

Comment:
I can experiment with this. I like using `?http_path=` because it's _very_ explicit. But since canonical URLs already include a `PATH` component I agree it makes as much sense to simply use that.